ql/src/test/resources/org/apache/hadoop/hive/ql/udf/generic/TestGenericUDFToUnixTimestampEvaluateStringString.csv

```diff
@@ -1,54 +1,128 @@
-1970-01-01 00:00:00;yyyy-MM-dd HH:mm:ss;Etc/GMT;DATETIME;0
```

Comment:
In order to make the tests easier to read and facilitate the diff I would suggest the following. Keep existing lines as they are and just add SMART at the end demonstrating that this value retains backward compatibility.

Add the new tests for invalid dates/patterns at the end of the file.

Also consider removing repetitive test cases such as the combination `yyyy with STRICT`. Showing once that this is invalid and returns null is sufficient. Moreover, if SMART returns null then I guess by definition STRICT will return null so these tests seem a bit redundant.

Reply:
Fixed